samza-core/src/main/java/org/apache/samza/context/JobContextMetadata.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.context;
+
+import org.apache.samza.job.model.JobModel;
+import org.apache.samza.system.StreamMetadataCache;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class JobContextMetadata {
```

Comment:
Apologies for missing this earlier: I just realized that the `objectRegistry` is specific to each task here, not the whole job. Therefore, could you please name this object (and its usages) to something that reflects the task scope? Maybe `InternalTaskContext`, since it is not part of the Samza public API.

Reply:
Done